internals/secrethub/audit.go

```diff
@@ -82,30 +129,198 @@ func (cmd *AuditCommand) run() error {
+	if !f.headerPrinted {
+		header := make([]string, len(f.columns))
+		for i, col := range f.columns {
+			header[i] = strings.ToUpper(col.name)
 		}
+		formattedHeader := f.formatRow(header)
+		_, err := f.writer.Write(formattedHeader)
+		if err != nil {
+			return err
+		}
+		f.headerPrinted = true
 	}
 
-	err = tabWriter.Flush()
-	if err != nil {
-		return err
+	formattedRow := f.formatRow(values)
+	_, err := f.writer.Write(formattedRow)
+	return err
+}
+
+// formatRow formats the given table row to fit the configured width by
+// giving each cell an equal width and wrapping the text in cells that exceed it.
+func (f *tableFormatter) formatRow(row []string) []byte {
+	columnWidths := f.columnWidths()
+	grid := f.rowToGrid(row, columnWidths)
+
+	strRes := strings.Builder{}
+	for _, row := range grid {
+		strRes.WriteString(strings.Join(row, "  ") + "\n")
 	}
+	return []byte(strRes.String())
+}
 
-	return nil
+// rowToGrid returns a the given row split over a matrix in which all columns have equal length.
+// Longer values are split over multiple cells and shorter (or empty) ones are padded with " ".
+func (f *tableFormatter) rowToGrid(row []string, columnWidths []int) [][]string {
+	maxLinesPerCell := f.lineCount(row, columnWidths)
+
+	grid := make([][]string, maxLinesPerCell)
+	for i := 0; i < maxLinesPerCell; i++ {
+		grid[i] = make([]string, len(row))
+	}
+
+	for i, cell := range row {
+		columnWidth := columnWidths[i]
+		lineCount := len(cell) / columnWidth
+		for j := 0; j < lineCount; j++ {
+			begin := j * columnWidth
+			end := (j + 1) * columnWidth
+			grid[j][i] = cell[begin:end]
+		}
+
+		charactersLeft := len(cell) % columnWidth
+		if charactersLeft != 0 {
+			grid[lineCount][i] = cell[len(cell)-charactersLeft:] + strings.Repeat(" ", columnWidth-charactersLeft)
+		} else if lineCount < maxLinesPerCell {
+			grid[lineCount][i] = strings.Repeat(" ", columnWidth)
+		}
+
+		for j := lineCount + 1; j < maxLinesPerCell; j++ {
+			grid[j][i] = strings.Repeat(" ", columnWidth)
+		}
+	}
+
+	return grid
+}
+
+// lineCount returns the number of lines the given table row will occupy after splitting the
+// cell values that exceed their column width.
+func (f *tableFormatter) lineCount(row []string, widths []int) int {
+	maxLinesPerCell := 1
+	for i, value := range row {
+		lines := len(value) / widths[i]
+		if len(value)%widths[i] != 0 {
+			lines++
+		}
+		if lines > maxLinesPerCell {
+			maxLinesPerCell = lines
+		}
+	}
+	return maxLinesPerCell
+}
+
+// columnWidths returns the width of each column based on their maximum widths
+// and the table width.
+func (f *tableFormatter) columnWidths() []int {
+	if f.computedColumnWidths != nil {
+		return f.computedColumnWidths
+	}
+	res := make([]int, len(f.columns))
+
+	// Distribute the maximum width equally between all columns and repeatedly
+	// check if any of them have a smaller maximum width and can be shrunk.
+	// Stop when no columns can be further adjusted.
```

Comment:
I'm having some trouble understanding what's happening in this function. Maybe it helps if these comments are moved to the place where they actually apply.